packages/components/src/public/QueryModel/GridPanel.tsx

```diff
@@ -728,6 +728,24 @@ export class GridPanel<T = {}> extends PureComponent<Props<T>, State> {
         });
     };
 
+    updateColumnTitle = (updatedCol: QueryColumn): void => {
+        const { model } = this.props;
+        this.saveAsSessionView({
+            columns: model.displayColumns
+                .map(col => {
+                    if (col.index === updatedCol.index) {
+                        return {
+                            fieldKey: updatedCol.index,
+                            title: updatedCol.caption === updatedCol.name ? '' : updatedCol.caption,
+                        }
+                    } else {
+                        return {fieldKey: col.index, title: col.caption === col.name ? '' : col.caption}
```

Comment:
looks like there are now 3 places that do this same type of thing. thoughts on factoring that out to a little mini helper function so that we don't have to keep duplicating that `col.caption === col.name` logic?

Reply:
Done. Added as a `get` method on `QueryColumn` itself.